### ISSUE TITLE
allow double quotes for continuation, even if quote-type is ‘single’

### ISSUE THREAD

**Issue by dato** [adrienverge/yamllint#753]
This is similar to #663, but not quite the same.

Consider:

```console
$ cat cont.yaml
a: "b\
    c"
```

Comparing yamllint 1.35 to 1.37, I see an improvement, since quotes are allowed now:

```console
$ yamllint --version && yamllint -d 'rules: {quoted-strings: {required: only-when-needed, quote-type: double}}' cont.yaml
yamllint 1.35.1
./cont.yaml
  1:4       error    string value is redundantly quoted with double quotes  (quoted-strings)

$ pip install yamllint==1.37.1
$ yamllint --version && yamllint -d 'rules: {quoted-strings: {required: only-when-needed, quote-type: double}}' cont.yaml
yamllint 1.37.1
```

However, if I change to using `quote-type: single`, yamllint complains that I’m using double quotes instead of single quotes:

```console
$ yamllint --version && yamllint -d 'rules: {quoted-strings: {required: only-when-needed, quote-type: single}}' cont.yaml              
yamllint 1.37.1
cont.yaml
  1:4       error    string value is not quoted with single quotes  (quoted-strings)
```

(To the best of my knowledge, backslash-as-continuation-char only works with double quotes, not single quotes. The sample _cont.yaml_ would typically be used for much longer strings, of course.)

Thanks for considering!

**Comment by adrienverge** [comment]
Hello @dato,

Thanks for this report.

I agree that *double quotes `"` + backslashes `\` at end of lines* are the only way to break a long string if it doesn't contain spaces. (Note: I sometimes re-read [this helpful post](https://stackoverflow.com/a/21699210) to remember multi-line strings features.)

E.g. we couldn't achieve the same without `"`:
```yaml
# "bc"
- "b\
   c"

# "b\\ c"
- 'b\
   c'

# "b c"
- >-
    b
    c

# "b\nc"
- |-
    b
    c
```

So I agree that we could modify yamllint to avoid reporting such cases.

However, yamllint rules `quoted-strings` and `line-length` don't communicate, which means `quoted-strings` doesn't know whether the line **needs** to be broken in 2 lines or not.
In other words, while it's obvious that the following use case is valid and shouldn't be reported:
```yaml
- "https://example.com/a/very/very/very/…/…/…/…/…/…/…/…/…/…/…/…/very/very\
   /long/URL"
```
… the following one could be written in only one line and use single quotes (`'bc'`), so we could argue it should be reported by rule `quoted-strings`:
```yaml
- "b\
   c"
```

If we change yamllint to fix the problem you described, the side effect would be to stop reporting such "short" strings that don't need double quotes because they could fit in one line.

What do you think? Would that be OK for all users?

**Comment by NightTsarina** [comment]
[Note: I am a colleague of @dato, so I am also affected by this]

@adrienverge While I agree that the second case you describe should probably be written in a single line, I'd say that if the user is breaking lines in this manner they probably have a reason for it and should not be reported?

**Comment by dato** [comment]
Hello @adrienverge! Thanks for your detailed response. 

I also agree with what @NightTsarina said above: it seems far more likely to use continuation lines to break long strings, so it makes sense to me to favor auto-detection there, at the cost of the shorter strings case.

Thanks again!
